geostructures/parsers.py

```diff
@@ -24,6 +25,242 @@
 }
 
 
+def _get_datetime_pandas(start_time, end_time):
+    """
+    Converts pandas Timestamps to Python datetime objects and returns a TimeInterval.
+
+    Args:
+        start_time (pd.Timestamp or None): The start time.
+        end_time (pd.Timestamp or None): The end time.
+
+    Returns:
+        TimeInterval: The time interval representing the start and end time.
+    """
+    import pandas as pd
+    from geostructures.time import TimeInterval
+
+    if pd.notnull(start_time) or pd.notnull(end_time):
```

Comment:
let's take a quick re-look at this block. If start_time/end_time is not null but is also not a timestamp it could cause issues